src/main/java/seedu/address/model/expense/Expense.java

```diff
@@ -0,0 +1,90 @@
+package seedu.address.model.expense;
+
+import static seedu.address.commons.util.CollectionUtil.requireAllNonNull;
+
+import java.time.LocalDate;
+import java.util.Objects;
+
+import seedu.address.commons.util.ToStringBuilder;
+
+/**
+ * Represents an Expense in the finance tracker.
+ * Guarantees: details are present and not null, field values are validated, immutable.
+ */
+public class Expense {
+    private final Name name;
+    private final Amount amount;
+    private final Category category;
+    private final LocalDate date;
```

Comment:
Would it be better to use LocalDateTime instead of LocalDate since we eventually have to store time information?